app/frontend/components/charts/PieChartComponent.tsx

```
@@ -0,0 +1,128 @@
+
+interface PieChartComponentProps {
+  data: FundRecord[] | undefined
+  title: string
+}
+
+export default function PieChartComponent ({ data, title }: PieChartComponentProps): React.ReactElement {
+  const chartData = preparePieData(data)
+
+  // Create config object for the chart
+  const chartConfig = chartData.reduce((acc, item) => ({
+    ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)